### ISSUE TITLE
[linux] Linux DeviceLayer needs WiFi ConnectivityMgr

### ISSUE THREAD

**Issue by turon** [project-chip/connectedhomeip#739]
#### Problem
The Linux device layer needs a ConnectivityMgr that handles a single, primary WiFi interface.

#### Proposed Solution
Implement the ConnectivityMgr with WiFi support.

Propose to interface directly to wpasupplicant.

Not in scope: conman, shill, or ubuntu / debian network manager.

Device layer assumes it is master of connectivity.  Target is embedded Linux type devices such as RasPi or smart speaker.

Include instructions for testing on a laptop or workstation, including instructions on how to disable local network mgmt and give control to CHIP app.

- Status - 2020-06-10
    - Draft PR posted -- DBUS client API to wpasupplicant implemented
    - No ProvisioningService integration yet in CHIP -- only tested in Weave environment thus far
    - DBUS using gmain_loop -- may cause conflicts with ThreadStack -- team to coord offline
    - Suggest unit tests based with credentials loaded from file
    - Script to launch required services such as wpasupplicant
    - Difficult to define AP in simulation w/o hardware
    - Evaluate using mac80211 hwsim module

**Comment by turon** [comment]
@kghost : Once #1084, only need a few days to land since PR already uses gio.
@yunhanw-google : How do you test WiFi provisioning?
@kghost : No tests right now -- simulating WiFi not trivial

**Comment by turon** [comment]
Status 2020-07-22:
- Was blocking dbus
- Have not had bandwidth to revisit since dbus PR merged
- reassigning to @yufengwangca

**Comment by turon** [comment]
Status 2020-08-05:
- 2 PRs on Wifi merged
- 1 more in review
- next to add shell commands to control and test wifi manager

**Comment by yufengwangca** [comment]
All PRs to support WiFi over GDBus are merged. Next to add shell commands to control and test wifi manager which is tracked by https://github.com/project-chip/connectedhomeip/issues/1696
